engine/items/inventory.asm

```diff
@@ -0,0 +1,658 @@
+	ld [hl], a ; store the new item count
+	ld [wItemQuantityBuffer], a
+	
+	scf ; set carry flag to indicate success
+	ret
+
+.fail
+	and a ; unset carry flag to indicate failure
+	ret
+
+CheckTMHM: ; 03:4CF4
+; Returns whether or not inventory contains TMHM in c
+	ld b, 0
+	ld hl, wTMsHMs
+	add hl, bc
+	ld a, [hl]
+	and a
+	ret z
+	scf
+	ret
+
+GetTMHMNumber: ; 03:4CFF
+; Return the item id of a TM/HM by number c.
+	ld a, c
+	ld c, 0 ; initialize the non-TM count to 0
+	
+	sub ITEM_TM01 ; a = TM index (if TM)
+	jr c, .notMachine
+	cp ITEM_C8 - ITEM_TM01 ; is the item ITEM_C8 ?
+	jr z, .notMachine ; If so, fail
+	jr c, .finish ; If the TM index is below, then finish
+	
+	inc c ; increase the non-TM count
+	cp ITEM_E1 - ITEM_TM01 ; is the item ITEM_E1?
+	jr z, .notMachine ; if so, fail
+	
+	jr c, .finish ; If the TM index is below, then finish
+	inc c ; otherwise, increase the non-TM count
+
+.finish
+	sub c ; subtract the amount of non TM-items that appear before this to get the true TM index
+	ld c, a ; store into c
+	
+	scf ; set the carry flag to indicate success
+	ret
+	
+.notMachine
+	and a ; unset the carry flag to indicate failure
+	ret
+	
+SECTION "_CheckTossableItem", ROMX[$53AD], BANK[$03]
+
+_CheckTossableItem:: ; 03:53AD
+; Return 1 in wItemAttributeParamBuffer and carry if wCurItem can't be removed from the bag.
+	ld a, ITEMATTR_PERMISSIONS
+	call GetItemAttr
+	bit CANT_TOSS_F, a
+	jr nz, ItemAttr_ReturnCarry
+	and a
+	ret
+
+CheckSelectableItem: ; 03:53B8
+; Return 1 in wItemAttributeParamBuffer and carry if wCurItem can't be selected.
+	ld a, ITEMATTR_PERMISSIONS
+	call GetItemAttr
+	bit CANT_SELECT_F, a
+	jr nz, ItemAttr_ReturnCarry
+	and a
+	ret
+
+CheckItemPocket: ; 03:53C3
+; Return the pocket for wCurItem in wItemAttributeParamBuffer.
+	ld a, ITEMATTR_POCKET
+	call GetItemAttr
+	and $0F
+	ld [wItemAttributeParamBuffer], a
+	ret
+
+CheckItemContext: ; 03:53CE
+; Return the context for wCurItem in wItemAttributeParamBuffer.
+	ld a, ITEMATTR_HELP
+	call GetItemAttr
+	and $0F
+	ld [wItemAttributeParamBuffer], a
+	ret
+
+CheckItemMenu: ; 03:53D9
+; Return the menu for wCurItem in wItemAttributeParamBuffer.
+	ld a, ITEMATTR_HELP
+	call GetItemAttr
+	swap a
+	and $f
+	ld [wItemAttributeParamBuffer], a
+	ret
+	
+GetItemAttr: ; 03:53E6
+; Get attribute a of wCurItem.
+	push hl
+	push bc
+	
+	ld hl, ItemAttributes
+	ld c, a ; c = item data byte index to retrieve
+	ld b, 0
+	add hl, bc ; hl = address for item
+	
+	xor a
+	ld [wItemAttributeParamBuffer], a ; set the item param value to 0
+	
+	ld a, [wCurItem] ; a = item id
+	dec a
+	ld c, a ; c = item id - 1
+	ld a, ITEMATTR_STRUCT_LENGTH ; size of each item attribute data set
+	call AddNTimes ; Go to the row for this item
+	
+	ld a, BANK( ItemAttributes )
+	call GetFarByte ; Get the byte from the table
+	
+	pop bc
+	pop hl
+	ret
+
+ItemAttr_ReturnCarry: ; 03:5405
+	ld a, 1
+	ld [wItemAttributeParamBuffer], a
+	scf
+	ret
+	
+GetItemPrice: ; 03:540C
+; Return the price of wCurItem in de.

```

Comment:
If you're going to put one comment after the corresponding label, why put all of the others before of it? I prefer putting them after the label, but I'm fine with anything as long as the same file is consistent.

Reply:
those were a direct copy from crystal.  I moved them to before the label to match other comments